tests/network/dry_run/test_dry_run_kubemacpool.py

```diff
@@ -130,11 +117,12 @@ def dry_run_vm_with_mac_address(
 def test_dry_run_mac_not_saved(
     dry_run_vma_mac_address,
     vm_with_mac_address,
-    running_vm_mac_addresses,
 ):
     """Assure that the MAC address given to a VM created in dry-run mode is still available in the MAC-pool."""
-    assert dry_run_vma_mac_address == running_vm_mac_addresses, (
-        f"{vm_with_mac_address.name} MAC address is {running_vm_mac_addresses}"
+    vm_interfaces = vm_with_mac_address.get_interfaces()
+    secondary_net_mac = vm_interfaces[1][MAC_ADDRESS]
```

Comment:
Are you sure kubevirt always places the default pod interface as the first one in the list, thus any secondary interface will necessarily start from index 1?
I really don't like the approach of relying on presumed indices. Maybe it would be better to iterate through the interfaces and get the one that its corresponding entry in `networks` is `multus`:
```
            networks:
            - name: default
              pod: {}
            - multus:
                networkName: lnx-br-nad
              name: br1
```
WDYT?

Reply:
> Generally speaking, if the default is not specified, nothing will add it implicitly to the VM spec template.

Which is one more reason to explicitly search for the `multus` interface. If the default is indeed not added, the the mutlus interface will occupy `vm_interfaces[0]`, and trying to access `vm_interfaces[1]` will end with a failure.

Reply:
You are right, but why do you request this on this change? This should be done all over.
And I prefer this is done by replacing all VMs to use the ones currently in libs/vm.

Reply:
Yes, you are actually right.
Now I see that referencing the specific index (1) in the interfaces list, instead of looking up the interface we are interested in (either by name or by type) is a legacy from before this change.
Very well, I will approve, and we will have this handling in all the places where an interface lookup is required, as we spoke.